pkg/config/validation/agent/validation.go

```diff
@@ -673,6 +676,50 @@ func ValidateWildcardDomain(domain string) error {
 	return nil
 }
 
+// ValidateWildcardDomainForVirtualServiceBoundToGateway checks that a domain is a valid FQDN, but also allows wildcard prefixes.
+// If it is an SNI domain, then it does a special validation where it allows a
+// which matches outbound_.<PORT>_._.<HOSTNAME> format
+func ValidateWildcardDomainForVirtualServiceBoundToGateway(sni bool, domain string) error {
+	if err := CheckDNS1123Preconditions(domain); err != nil {
+		return err
+	}
+
+	// check if its an auto generated domain, with outbound_ as a prefix.
+	if sni && strings.HasPrefix(domain, outboundHostPrefix) {
```

Comment:
I think this would fail for `outbound.com` which is valid? likely we need `outboundHostPrefix` to be `outboundHostPrefix_.`?

Reply:
fixed.